harness/determined/pytorch/_trainer.py

```diff
@@ -0,0 +1,211 @@
+import contextlib
+import logging
+import random
+from typing import Dict, Optional, Union, cast
+
+import torch
+import torch.distributed as dist
+
+import determined as det
+from determined import core, horovod, pytorch
+from determined.horovod import hvd
+from determined.profiler import DummyProfilerAgent, ProfilerAgent
+
+
+class Trainer:
+    def __init__(self, trial: pytorch.PyTorchTrial, context: pytorch.PyTorchTrialContext):
+        self._trial = trial
+        self._context = context
+        self._cluster_info = det.get_cluster_info()
+        self._core = self._context._core
+        self._distributed_backend = det._DistributedBackend()
+        self._det_profiler = DummyProfilerAgent()
+        self._trial_controller = None
+        self._local_training = self._cluster_info is None
+
+    def configure_profiler(
+        self, sync_timings: bool, enabled: bool, begin_on_batch: int, end_after_batch: int
+    ):
+        assert self._cluster_info, "Determined profiler must be run on cluster"
+        self._det_profiler = ProfilerAgent(
+            trial_id=str(self._cluster_info.trial.trial_id),
+            agent_id=self._cluster_info.agent_id,
+            master_url=self._cluster_info.master_url,
+            profiling_is_enabled=enabled,
+            global_rank=self._core.distributed.get_rank(),
+            local_rank=self._core.distributed.get_local_rank(),
+            begin_on_batch=begin_on_batch,
+            end_after_batch=end_after_batch,
+            sync_timings=sync_timings,
+        )
+
+    def train(
+        self,
+        max_epochs: Optional[int] = None,
+        # OR
+        max_batches: Optional[int] = None,
+        min_checkpoint_period: Union[pytorch.TrainUnit, int] = 1,
+        min_validation_period: Union[pytorch.TrainUnit, int] = 1,
```

Comment:
I thought we were naming this just "checkpoint_period" and "validation_period"?  Why bother with these (wrong) `min_` prefixes?

Reply:
changed.